File: core/src/main/java/cz/dynawest/jtexy/modules/TableModule.java

package cz.dynawest.jtexy.modules;


import cz.dynawest.jtexy.parsers.TexyEventListener;
import java.util.logging.*;


/**
 *
 * @author Ondrej Zizka
 */
public class TableModule extends TexyModule
{
  private static final Logger log = Logger.getLogger( TableModule.class.getName() );

	private PatternHandler somePH = null;


	// -- Module meta-info -- //

	@Override public TexyEventListener[] getEventListeners() {
		return new TexyEventListener[]{};
	}

	@Override	protected PatternHandler getPatternHandlerByName(String name) {
		if( somePH.getName().equals(name) ) {
			return somePH;
		} else {
			return null;
		}
	}

}// class HtmlModule

File: core/src/main/java/cz/dynawest/jtexy/modules/TableModule.kt
package cz.dynawest.jtexy.modules

import cz.dynawest.jtexy.events.TexyEvent
import cz.dynawest.jtexy.events.TexyEventListener
import java.util.logging.Logger

/**
 *
 * @author Ondrej Zizka
 */
class TableModule : TexyModule() {
    private val somePH: PatternHandler? = null
    override val eventListeners: List<TexyEventListener<TexyEvent>>
        get() = emptyList()

    override fun getPatternHandlerByName(name: String): PatternHandler? {
        return if (somePH?.name == name) {
            somePH
        } else {
            null
        }
    }

    companion object {
        private val log = Logger.getLogger(TableModule::class.java.name)
    }
}
